modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java

```diff
@@ -358,8 +358,9 @@ public CommerceOrder addOrUpdateCommerceOrder(
 			commerceOrderLocalService.updatePaymentStatus(
```

Comment:
This could be an issue with MVCC, we are updating the same order's persistence 2 times in a row, without assigning the result after the first call, so that the second update is done on an older version of the order.

Reply:
In general I don't like to call other services from a method body, we should try to avoid that. Since we are already updating the order's persistence, we can just set the values we want directly within this method.